### PR TITLE
Feature - Implement login service

### PR DESCRIPTION
**Pull request by mvn-daiho-dn**

### CODE REVIEW

File: Demo/Library/Ext/String.swift [hunk 1]
```diff
@@ -9,7 +9,15 @@
 import Foundation
 
 extension String {
+
```
Comment (at-daonguyen) [inline Demo/Library/Ext/String.swift]: Please remove whitespace.

File: Demo/Library/Ext/String.swift [hunk 2]
```diff
@@ -9,7 +9,15 @@
 import Foundation
 
 extension String {
+    
     var host: String? {
         return (try? asURL())?.host
     }
+
```
Comment (at-daonguyen) [inline Demo/Library/Ext/String.swift]: Please remove whitespace.

File: Demo/Model/API/UserAPIs.swift [hunk 3]
```diff
@@ -7,14 +7,25 @@
 //
 
 import Foundation
+import Alamofire
 
 extension API.User {
+
```
Comment (at-daonguyen) [inline Demo/Model/API/UserAPIs.swift]: Please remove whitespace.

File: Demo/Model/API/UserAPIs.swift [hunk 4]
```diff
@@ -7,14 +7,25 @@
...
+    
     struct LoginParams {
         let mail: String
         let pass: String
     }
 
     func login(params: LoginParams, completion: @escaping Completion) {
-
+        var headers: [String: String] = [:]
+        if let base64Encode = "\(params.mail):\(params.pass)".base64Encode {
```
Comment (at-daonguyen) [inline Demo/Model/API/UserAPIs.swift]: Can you separate this operation to new function?

**Review by at-daonguyen — CHANGES_REQUESTED** [review]
@at-daiho Please check my comments.

File: Demo/Model/API/UserAPIs.swift [hunk 5]
```diff
@@ -7,14 +7,25 @@
...
+            if !base64Encode.isEmpty {
+                headers["Authorization"] = "Basic \(base64Encode)"
+            }
+        }
+        
+        let _ = APIManager.request(method: HTTPMethod.get, urlString: ApiPath.Users, parameters: nil, headers: headers) { (result) in
```
Comment (ghost) [inline Demo/Model/API/UserAPIs.swift]: I think we already have `let api = APIManager()` in `APIManager.swift`

Comment (mvn-daiho-dn) [reply Demo/Model/API/UserAPIs.swift]: request is static func.

Comment (ghost) [reply Demo/Model/API/UserAPIs.swift]: you could change it, right? @at-daonguyen  👯‍♂️

File: Demo/Model/API/UserAPIs.swift [hunk 6]
```diff
@@ -7,14 +7,25 @@
...
 
     func login(params: LoginParams, completion: @escaping Completion) {
+        var headers: [String: String] = [:]
+        if let base64Encode = "\(params.mail):\(params.pass)".base64Encode {
+            if !base64Encode.isEmpty {
+                headers["Authorization"] = "Basic \(base64Encode)"
+            }
+        }
 
+        let _ = api.request(method: HTTPMethod.get, urlString: ApiPath.Users, parameters: nil, headers: headers) { (result) in
```
Comment (ghost) [inline Demo/Model/API/UserAPIs.swift]: @at-daiho I think we no need `let` here 👯  hehe

**Comment by at-daonguyen** [comment]
@at-daiho Any update?

**Review by at-daonguyen — APPROVED** [review]

**Pull request merged**